demos/jans-tarp/src/popup.js

```diff
@@ -56,6 +57,52 @@ const qs = require('qs');
     });
   }
 
+  function customLaunchWebAuthFlow(options, callback) {
+    var requestId = Math.random().toString(36).substring(2);
+    var authUrl = options.url + "&state=" + requestId;
+    chrome.tabs.create({ url: authUrl }, function (tab) {
+      var intervalId = setInterval(function () {
+        chrome.tabs.get(tab.id, function (currentTab) {
+          if (!currentTab) {
+            clearInterval(intervalId);
+            chrome.tabs.remove(tab.id);
+            callback(undefined, new Error('Authorization tab was closed.'));
+          }
+        });
+      }, 1000);
+
+      chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+        if (!!chrome.runtime.lastError) {
+          chrome.tabs.remove(tabId);
+          callback(undefined, chrome.runtime.lastError);
+        }
+        if (tabId === tab.id && changeInfo.status === "complete") {
```

Comment:
Please check if **changeInfo** object have **status** key or not before applying status condition by using "?". same for tab.id as well.
For example add like this :-
 if (tabId === tab?.id && changeInfo?.status === "complete")

Reply:
done